### ISSUE TITLE
train on dtu with different numdepth and interval

### ISSUE THREAD

**Issue by fariba87** [maybeLx/MVSFormerPlusPlus#36]
thank you for your great work.
i want to finetune the model on the same dtu dataset but with different numdepth (192->512) and intervalscale(1.06->0.25). as i want to get inference from the model for a high detail object which depth variation is a lot on its surface i think 192 may not suffice. so i decide to change these parameters in config file and finetune the model.(i want to finetune in small set of dtu scans). you saved the checkpoint in epoch 6, although after finetuning the loss decrease from 16 to ~6 but it does not decrease anymore. what is your suggestion for my problem?

**Comment by fariba87** [comment]
besides, i want to know if i change depth_type from ce to reg it has positive effect? and also to increase "ndepths": [  32,  16,     8,    4 ], from 4 stages to 5 stages?

**Comment by maybeLx** [comment]
f you modify the `ndepths` and `ndepths` parameters, you’ll notice that `ndepths` does not affect the interval in stage 1. (For clarity in comparison, you can set `inverse_depth` to `False`.) To verify this, you can print the `depth_samples` at this [line of code]
(https://github.com/maybeLx/MVSFormerPlusPlus/blob/bdd980cfbae9a863b5489e2ed973612b192d11a5/models/networks/DINOv2_mvsformer_model.py#L150).

**Comment by maybeLx** [comment]
If you modify the `intervalscale` and `ndepth`s parameters, you should print the `depth_samples` in `stage 1` to verify that the `depth_samples` includes the range 425–935 (the ground truth depth in DTU is approximately 425–935).

**Comment by maybeLx** [comment]
> besides, i want to know if i change depth_type from ce to reg it has positive effect? and also to increase "ndepths": [ 32, 16, 8, 4 ], from 4 stages to 5 stages?

As mentioned in our [MVSFormer](https://github.com/ewrfcas/MVSFormer), CE performs better than REG. The confidence with CE is significantly better than with REG (You can use confidence of CE to filter points much easier ). You can also change the stage number and increase the ndepths. However, the most important thing is to ensure that you need to adjust the extrinsic and intrinsic parameters accordingly.

**Comment by fariba87** [comment]
thank you.so, if i understand well, maybe increasing the number stages(and adjusting camera parameters accordingly) and ndepths and retrain the model, may help to capture the depth with more detail?

**Comment by maybeLx** [comment]
We have not attempted to use more stages for training the model, as adding additional stages would reduce the image resolution in the coarse stage. A lower resolution in Stage 1 could negatively impact the accuracy of depth predictions. Since we  use the coarse-to-fine architecture. this could lead to error accumulation.
